src/effects/snowfall.c

```diff
@@ -1,13 +1,54 @@
 #include "common.h"
 #include "effects_internal.h"
 
+
+void snowfall_init(EffectInstance* effect);
+void snowfall_update(EffectInstance* effect);
+void snowfall_render(EffectInstance* effect);
 void snowfall_appendGfx(void* effect);
 
 INCLUDE_ASM(s32, "effects/snowfall", func_E008A000);
 
-INCLUDE_ASM(s32, "effects/snowfall", snowfall_main);
+void func_E008A000(SnowfallFXData* data);
+
+EffectInstance* snowfall_main(s32 arg0, s32 arg1) {
+    EffectBlueprint effectBp;
+    EffectInstance* effect;
+    SnowfallFXData* data;
+    s32 index;
+    s32 temp_s0;
```

Comment:
name this one numParts

Reply:
also can it be initialized when declared like in most effect functions?

Reply:
yeah, I'm not sure why I did that. I think that this was my first effects main.